chain/chain/src/chain.rs

```diff
@@ -905,6 +905,9 @@ impl Chain {
     {
         let block_hash = *block.hash();
         let timer = metrics::BLOCK_PROCESSING_TIME.start_timer();
+        let instant = Instant::now();
+        // We measure time once, because the histogram measures time of all block processing
+        // attempts, and filtering needs to be done manually.
```

Comment:
Oh, one more version which think would definitelly be strinctly better than the current one, as it fires up only one timer: 

```diff
diff --git a/chain/chain/src/chain.rs b/chain/chain/src/chain.rs
index fdfe9335b..31afcf71f 100644
--- a/chain/chain/src/chain.rs
+++ b/chain/chain/src/chain.rs
@@ -905,7 +905,6 @@ impl Chain {
     {
         let block_hash = *block.hash();
         let timer = metrics::BLOCK_PROCESSING_TIME.start_timer();
-        let instant = Instant::now();
         // We measure time once, because the histogram measures time of all block processing
         // attempts, and filtering needs to be done manually.
         let res = self.process_block_single(
@@ -917,10 +916,10 @@ impl Chain {
             block_orphaned_with_missing_chunks,
             on_challenge,
         );
-        timer.observe_duration();
+        let t = timer.stop_and_record();
         if res.is_ok() {
             metrics::BLOCK_PROCESSED_SUCCESSFULLY_TOTAL.inc();
-            metrics::BLOCK_SUCCESSFUL_PROCESSING_TIME.observe(instant.elapsed().as_secs_f64());
+            metrics::BLOCK_SUCCESSFUL_PROCESSING_TIME.observe(t);
 
             if let Some(new_res) = self.check_orphans(
                 me,
```

Reply:
Awesome, thanks